Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Spawnable/PrefabDocument.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/RTTI/TypeInfo.h>
+#include <AzCore/std/smart_ptr/unique_ptr.h>
+#include <AzCore/std/string/string.h>
+#include <AzToolsFramework/Prefab/Instance/Instance.h>
+#include <AzToolsFramework/Prefab/PrefabDomTypes.h>
+#include <AzToolsFramework/Prefab/Spawnable/EntityAliasTypes.h>
+#include <AzToolsFramework/Prefab/Spawnable/SpawnableUtils.h>
+
+namespace AzToolsFramework::Prefab::PrefabConversionUtils
+{
+    class PrefabProcessorContext;
+
+    class PrefabDocument final
+    {
+    public:
+        explicit PrefabDocument(AZStd::string name);
+        PrefabDocument(const PrefabDocument&) = delete;
+        PrefabDocument(PrefabDocument&&) = default;
+
+        PrefabDocument& operator=(const PrefabDocument&) = delete;
+        PrefabDocument& operator=(PrefabDocument&&) = default;
+
+        bool SetPrefabDom(const PrefabDom& prefab);
+        bool SetPrefabDom(PrefabDom&& prefab);
+
+        const AZStd::string& GetName() const;
+        const PrefabDom& GetDom() const;
+        PrefabDom&& TakeDom();
+
+        template<typename Component>
+        void ListEntitiesWithComponentType(const AZStd::function<bool(AzToolsFramework::Prefab::AliasPath&&)>& callback) const;
+        void ListEntitiesWithComponentType(
+            AZ::TypeId componentType, const AZStd::function<bool(AzToolsFramework::Prefab::AliasPath&&)>& callback) const;
```

Comment:
Should we make this a private member function?

Reply:
This function is meant to be called by users of `PrefabDocument`. The reason there is no use case for this in this PR is because the first customer is Networking, but switching to using this functions and Spawnable Entity Aliases also involves making changes on the Networking side, which will come later.